src/xml_parser_main_metadata_together.c

```diff
@@ -334,136 +328,227 @@ int cr_xml_parse_main_metadata_together(const char *primary_path,
                                         void *pkgcb_data,
                                         cr_XmlParserWarningCb warningcb,
                                         void *warningcb_data,
-                                        gboolean allow_out_of_order,
                                         GError **err)
 {
-    int ret = CRE_OK;
-    CR_FILE *primary_f = NULL;
-    CR_FILE *filelists_f = NULL;
-    CR_FILE *other_f = NULL;
-    GError *tmp_err = NULL;
+    assert(pkgcb || newpkgcb);
+    cr_PkgIterator* pkg_iterator = cr_PkgIterator_new(
+        primary_path, filelists_path, other_path, newpkgcb, newpkgcb_data, warningcb, warningcb_data, err
+    );
 
-    cr_CbData cbdata;
-    cbdata.in_progress_pkgs_list = NULL;
-    cbdata.in_progress_pkgs_hash = NULL;
-    cbdata.newpkgcb = newpkgcb;
-    cbdata.newpkgcb_data = newpkgcb_data;
-    cbdata.pkgcb = pkgcb;
-    cbdata.pkgcb_data = pkgcb_data;
+    assert(pkg_iterator);
```

Comment:
This `assert` is causing unittest fails, because the error checking is below it.